### ISSUE TITLE
Workspace not loading

### ISSUE THREAD

**Issue by hm2075** [gitpod-io/gitpod#2028]
Hi

using later builds found at
https://eu.gcr.io/gitpod-core-dev/build

hopefully this is the last of the issues

Warning  FailedScheduling  21s   workspace-scheduler  no suitable node found: Zero nodes available!

I have labelled accordingly

  gitpod.io/workload_meta: "true"
    gitpod.io/workload_workspace: "true"

**Comment by jgallucci32** [comment]
Do the following dynamic labels generated by Gitpod appear on the nodes?
```
gitpod.io/theia.0.5.0=available
gitpod.io/ws-sync=available
```

**Comment by hm2075** [comment]
beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    gitpod.io/diskPressure: "true"
    gitpod.io/theia.master.109: available
    gitpod.io/workload_meta: "true"
    gitpod.io/workload_workspace: "true"
    gitpod.io/ws-sync: available
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: xxx
    kubernetes.io/os: linux
    node-role.kubernetes.io/master: ""

**Comment by hm2075** [comment]
cleansed the system,  removed orphaned workspaces etc, 



tracking ws-sync

{"error":"rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.100.202.176:22999: connect: connection refused\"","instanceId":"94d79c2d-da55-46fd-8a24-e8c8f37c1296","message":"backup canary unavailable - maybe because of workspace shutdown","serviceContext":{"service":"ws-sync","version":""},"severity":"debug","time":"2020-10-20T15:04:40Z","userId":"6a4ccd19-66d2-436a-a1dd-3ca07aaf51b5","workspaceId":"fb588872-6cdf-495e-8364-8647de54562a"}

**Comment by hm2075** [comment]
i see the error


`{
  "location": "/mnt/workingarea/122a0f82-f811-4506-8da5-27d29aeab6d1/haxe-gitpod",
  "message": "Running git clone on workspace",
  "serviceContext": {
    "service": "ws-sync",
    "version": ""
  },
  "severity": "debug",
  "stage": "init",
  "time": "2020-10-20T15:09:48Z"
}{
  "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
  "containerID": "0bac5f3f58eff02b2d6e0a15de44e34070a8682bb47a781fa95e8cb8e402030f",
  "containerImage": "",
  "error": "not found\ngithub.com/containerd/containerd/errdefs.init\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.3.3/errdefs/errors.go:45\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5625\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.main\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:191\nruntime.goexit\n\t/home/gitpod/sdk/go1.15/src/runtime/asm_amd64.s:1374\ncontainer \"0bac5f3f58eff02b2d6e0a15de44e34070a8682bb47a781fa95e8cb8e402030f\" in namespace \"k8s.io\"\ngithub.com/containerd/containerd/errdefs.FromGRPC\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.3.3/errdefs/grpc.go:107\ngithub.com/containerd/containerd.(*remoteContainers).Get\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.3.3/containerstore.go:50\ngithub.com/gitpod-io/gitpod/ws-sync/pkg/cri.(*ContainerdCRI).handleContainerdEvent\n\t/tmp/build/components-ws-sync--app.be17edd28fbd915db71942934270b8b2baedc190/pkg/cri/containerd.go:133\ngithub.com/gitpod-io/gitpod/ws-sync/pkg/cri.(*ContainerdCRI).start.func1\n\t/tmp/build/components-ws-sync--app.be17edd28fbd915db71942934270b8b2baedc190/pkg/cri/containerd.go:117\nruntime.goexit\n\t/home/gitpod/sdk/go1.15/src/runtime/asm_amd64.s:1374",
  "message": "cannot find container we just received a create event for",
  "serviceContext": {
    "service": "ws-sync",
    "version": ""
  },
  "severity": "warning",
  "time": "2020-10-20T15:09:49Z"
}`

**Comment by jgallucci32** [comment]
@hm2075 I saw the same error `cannot find container we just received a create event for` and dug a little deeper to find out the Theia supervisor is failing to launch and shuts down. Suggest looking at the error described in https://github.com/gitpod-io/gitpod/issues/2029 to see if it matches your case. In my case I was able to fix it by correcting the ws-sync pod networking as described in the issue.

**Comment by hm2075** [comment]
ws-sync has gone in the later build,  check out the latest master branch       

only realised  after i made a proper school boy error rm -rf         


now ive got 

Last run a few seconds ago: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"/theia/supervisor\": stat /theia/supervisor: no such file or directory": unknow


i think i've seen this error logged before  here

**Comment by hm2075** [comment]
@AlexTugarev @corneliusludmann 

comes down to this


`[
  ws-daemon-kzjw5
]{
  "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
  "ID": "bb6d3643577da5abd06a1aa6c77d7ac2f77dd6bd7ed030bab5c6e35329376aec",
  "containerImage": "",
  "error": "not found\ngithub.com/containerd/containerd/errdefs.init\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.4.1/errdefs/errors.go:45\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5625\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.doInit\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:5620\nruntime.main\n\t/home/gitpod/sdk/go1.15/src/runtime/proc.go:191\nruntime.goexit\n\t/home/gitpod/sdk/go1.15/src/runtime/asm_amd64.s:1374\ncontainer \"bb6d3643577da5abd06a1aa6c77d7ac2f77dd6bd7ed030bab5c6e35329376aec\" in namespace \"k8s.io\"\ngithub.com/containerd/containerd/errdefs.FromGRPC\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.4.1/errdefs/grpc.go:107\ngithub.com/containerd/containerd.(*remoteContainers).Get\n\t/workspace/go/pkg/mod/github.com/containerd/containerd@v1.4.1/containerstore.go:50\ngithub.com/gitpod-io/gitpod/ws-daemon/pkg/container.(*Containerd).handleContainerdEvent\n\t/tmp/build/components-ws-daemon--app.e39981c5d4b87655cd4b2c581f4995a95498f292/pkg/container/containerd.go:162\ngithub.com/gitpod-io/gitpod/ws-daemon/pkg/container.(*Containerd).start\n\t/tmp/build/components-ws-daemon--app.e39981c5d4b87655cd4b2c581f4995a95498f292/pkg/container/containerd.go:145\nruntime.goexit\n\t/home/gitpod/sdk/go1.15/src/runtime/asm_amd64.s:1374",
  "message": "cannot find container we just received a create event for",
  "serviceContext": {
    "service": "ws-daemon",
    "version": ""
  },
  "severity": "warning",
  "time": "2020-10-21T06:41:28Z"
}[
 `



plz advise

**Comment by jankeromnes** [comment]
Hi @hm2075! Sorry about this issue. It probably comes from the fact that `ws-daemon` is a brand new component that we haven't had the opportunity to extensively test yet.

In fact, using the latest available commit in https://github.com/gitpod-io/gitpod is not something we support at this time (i.e. you're free to try that, but you may run into all sorts of problems, and we won't be able to effectively help you because we're also running into the same problems and are currently in the process of figuring them out).

The safer route is to use the latest release of Gitpod Self-Hosted, which is `0.5.0`. I believe you can get it like so:

```
$ helm repo add gitpod https://charts.gitpod.io
$ helm repo update
$ helm install gitpod gitpod/gitpod --timeout 60m --version 0.5.0 --values values.yaml
```

... with a proper `values.yaml` ([source](https://community.gitpod.io/t/issue-installing-on-self-managed-kubernetes/1705/2?u=jan)).

(We're also currently working on a good release process to publish new Gitpod Self-Hosted versions as frequently as possible, but also put these new versions through a robust testing & quality assurance process first to always guarantee a smooth install/upgrade experience.)

**Comment by hm2075** [comment]
hi @jankeromnes 
no need to be sorry,   I am in no rush at this moment in time to move this into production,

i moved to the later builds because of a fix (session:fix)    that isnt available in 0.5.0


is there a requirement to have a seperate gitpod self hosted version?         your made codebase is pretty good,  shouldnt you just look to provide instructions on that? otherwise won't you have too many places to maintain?     (gitlab are pretty good at this  - monthly release cycles etc)

I managed to figure out more or less everything on how to get the core build going on a clean centos8 box,     I have made plenty of notes so it can be replicated.  Everything from installing kubernetes, installing mysql,  ssl certs for gitpod,  setting up a registry etc which I will share.


for those behind corporate firewalls and a need to keep everything on site,   selfhosted versions of software such as gitpod, gitlab are much needed.  

There's obviously bigger hurdles for self hosted enterprise gitpod,   this includes using self cert,    proxy issues such as this https://github.com/gitpod-io/gitpod/issues/2002   etc etc

could we work together on this?  is there a better way to communicate and help test your later builds?
